mediatimestamp/__init__.py

```diff
@@ -22,962 +22,36 @@
-        return self.__class__(toff.sec, toff.ns, toff.sign)
-
-    def __rmul__(self, anint):
-        toff = TimeOffset(self.sec, self.ns, self.sign)
-        toff *= anint
-        return self.__class__(toff.sec, toff.ns, toff.sign)
-
-    def __truediv__(self, anint):
-        toff = type(self)(self.sec, self.ns, self.sign)
-        abs_anint = abs(anint)
-        sec = self.sec // abs_anint
-        ns = int((self.ns + (self.sec % abs_anint) * self.MAX_NANOSEC) / abs_anint + 5e-10)
-
-        toff.sec = sec + ns // self.MAX_NANOSEC
-        toff.ns = ns % self.MAX_NANOSEC
-        if anint < 0:
-            toff.sign *= -1
-        toff._make_valid()
-        return toff
-
-    def __floordiv__(self, anint):
-        return self.__truediv__(anint)
-
-    def __imul__(self, anint):
-        abs_anint = abs(anint)
-        ns_sec = self.ns * (abs_anint // self.MAX_NANOSEC)
-        ns = self.ns * (abs_anint % self.MAX_NANOSEC)
-
-        self.sec = self.sec * abs_anint + ns_sec + ns // self.MAX_NANOSEC
-        self.ns = ns % self.MAX_NANOSEC
-        if anint < 0:
-            self.sign *= -1
-        self._make_valid()
-        return self
-
-    def _complete_iadd_or_isub(self, sec, ns):
-        if ns >= self.MAX_NANOSEC:
-            ns -= self.MAX_NANOSEC
-            sec += 1
-        elif -ns >= self.MAX_NANOSEC:
-            ns += self.MAX_NANOSEC
-            sec -= 1
-        if sec < 0 and ns > 0:
-            ns -= self.MAX_NANOSEC
-            sec += 1
-        elif sec > 0 and ns < 0:
-            ns += self.MAX_NANOSEC
-            sec -= 1
-
-        if sec < 0 or ns < 0:
-            self.sign = -1
-            self.sec = abs(sec)
-            self.ns = abs(ns)
-        else:
-            self.sign = 1
-            self.sec = sec
-            self.ns = ns
-        self._make_valid()
-
-    def _get_fractional_seconds(self, fixed_size=False):
-        div = self.MAX_NANOSEC / 10
-        rem = self.ns
-        sec_frac = ""
-
-        for i in range(0, 9):
-            if not fixed_size and i > 0 and rem == 0:
-                break
-
-            sec_frac += '%i' % (rem / div)
-            rem %= div
-            div /= 10
-
-        return sec_frac
-
-    def _cast_arg(self, other):
-        if isinstance(other, integer_types):
-            return TimeOffset(other)
-        elif isinstance(other, float):
-            return TimeOffset.from_sec_frac(str(other))
-        else:
-            return other
-
-    def _make_valid(self):
-        if self.sign >= 0 or (self.sec == 0 and self.ns == 0):
-            self.sign = 1
-        else:
-            self.sign = -1
-        if self.sec < 0:
-            self.sec = 0
-            self.ns = 0
-            self.sign = 1
-        elif self.ns < 0:
-            self.ns = 0
-        elif self.ns >= self.MAX_NANOSEC:
-            self.ns = self.MAX_NANOSEC - 1
-
-
-class Timestamp(TimeOffset):
-    """A nanosecond precision timestamp."""
+class BaseTimeOffset (object):
```

Comment:
Would this (and the other one_ benefit from a docstring explaining what they're for?